include/inviwo/core/properties/listproperty.h

```diff
@@ -0,0 +1,207 @@
+                 InvalidationLevel invalidationLevel = InvalidationLevel::InvalidResources,
+                 PropertySemantics semantics = PropertySemantics::Default);
+    ListProperty(const ListProperty& rhs);
+    ListProperty& operator=(const ListProperty& that);
+    virtual ListProperty* clone() const override;
+    virtual ~ListProperty() = default;
+
+    virtual std::string getClassIdentifierForWidget() const override;
+
+    virtual void set(const Property* src) override;
+    void set(const ListProperty* src);
+
+    /**
+     * \brief set the max number of list elements. This will remove additional properties if the
+     * list property contains more than \p n items.
+     *
+     * @param n    maximum number of elements in this list property
+     */
+    void setMaxNumberOfElements(size_t n);
+    size_t getMaxNumberOfElements() const;
+
+    /** 
+     * \brief remove all list entries
+     */
+    void clear();
+
+    /**
+     * \brief add a list entry which is created from the respective prefab object.
+     * This function has no effect if the list size will exceed the maximum number of elements.
+     *
+     * @param prefabIndex   index of prefab object used for creating the new entry
+     * @throw RangeException  in case prefabIndex is invalid
+     */
+    void addProperty(size_t prefabIndex);
+
+    /**
+     * \brief add \p property as new list entry. The type of the property must match one of the
+     * prefab objects. This function has no effect if the list size will exceed the maximum number
+     * of elements.
+     *
+     * @param property     property to be added
+     * @param owner        if true, the list property takes ownership of the property
+     * @throw Exception    if the type of \p property does not match any prefab object
+     */
+    virtual void addProperty(Property* property, bool owner = true) override;
+
+    /**
+     * \brief add \p property as new list entry. The type of the property must match one of the
+     * prefab objects. This function has no effect if the list size will exceed the maximum number
+     * of elements.
+     *
+     * @param property     property to be added
+     * @throw Exception    if the type of \p property does not match any prefab object
+     */
+    virtual void addProperty(Property& property) override;
+
+    virtual Property* removeProperty(const std::string& identifier) override;
+    virtual Property* removeProperty(Property* property) override;
+    virtual Property* removeProperty(Property& property) override;
+    virtual Property* removeProperty(size_t index) override;
+
+    /**
+     * \brief return number of prefab objects
+     *
+     * @return count of prefabs
+     */
+    size_t getPrefabCount() const;
+
+    /**
+     * \brief add a new prefab object \p p to be used as template when instantiating new list
+     * elements
+     *
+     * @param p  prefab object
+     */
+    void addPrefab(std::unique_ptr<Property>&& p);
```

Comment:
You don't need the && here, can just pass it by value...

Reply:
fixed